### PR TITLE
Skeleton perl installdirs vendor

### PR DESCRIPTION
**Pull request by mbargull**
In https://github.com/conda-forge/perl-feedstock/pull/49 I'm changing the directory layout for conda-forge's Perl packages to have the `vendor` dirs split off so that we (as in Conda channel maintainers) can act as the "vendor" and have the Perl packages we build in the "vendor" location and leave the "site" location to the user's (non-Conda) packages.

### CODE REVIEW

File: conda_build/skeletons/cpan.py [hunk 1]
```diff
@@ -346,7 +346,8 @@ def get_core_modules_for_this_perl_version(version, cache_dir):
 # meta_cpan_url="http://api.metacpan.org",
 def skeletonize(packages, output_dir=".", version=None,
                 meta_cpan_url="https://fastapi.metacpan.org/v1",
-                recursive=False, force=False, config=None, write_core=False):
+                recursive=False, force=False, config=None, write_core=False,
+                installdirs="vendor"):
```
Comment (isuruf) [inline conda_build/skeletons/cpan.py:350]: This should be taken from `config` right?. Otherwise the CLI value is not used.

Comment (mbargull) [reply conda_build/skeletons/cpan.py:350]: Thanks for the review!
There is some indirection in how the CLI invokes `skeletonize`:
https://github.com/mbargull/conda-build/blob/bcb43db98f730ee92f7123b9caf897cc8230e171/conda_build/cli/main_skeleton.py#L60-L61
https://github.com/mbargull/conda-build/blob/bcb43db98f730ee92f7123b9caf897cc8230e171/conda_build/api.py#L268-L269
and `Config` accepts any non-listed key as attributes to itself:
https://github.com/mbargull/conda-build/blob/bcb43db98f730ee92f7123b9caf897cc8230e171/conda_build/config.py#L305-L307
In other words: If the CLI argument and `skeletonize` parameter have the same name, this works implicitly. Like with the `meta_cpan_url` for example.
